datasets/bnl_processed_newspapers/bnl_processed_newspapers.py

```diff
@@ -0,0 +1,186 @@
+            {
+                "ark_id": datasets.Value("string"),
+                "source": datasets.Value("string"),
+                "url": datasets.Value("string"),
+                "title": datasets.Value("string"),
+                "ispartof": datasets.Value("string"),
+                "text": datasets.Value("string"),
+                "pub_date": datasets.Value("timestamp[s]"),
+                "publisher": datasets.Value("string"),
+                "article_type": datasets.ClassLabel(
+                    names=[
+                        "ADVERTISEMENT_SECTION",
+                        "BIBLIOGRAPHY",
+                        "CHAPTER",
+                        "INDEX",
+                        "CONTRIBUTION",
+                        "TABLE_OF_CONTENTS",
+                        "WEATHER",
+                        "SHIPPING",
+                        "SECTION",
+                        "ARTICLE",
+                        "TITLE_SECTION",
+                        "DEATH_NOTICE",
+                        "SUPPLEMENT",
+                        "TABLE",
+                        "ADVERTISEMENT",
+                        "CHART_DIAGRAM",
+                        "ILLUSTRATION",
+                        "ISSUE",
+                    ]
+                ),
+                "extent": datasets.Value("int32"),
+            }
+        )
+        return datasets.DatasetInfo(
+            # This is the description that will appear on the datasets page.
+            description=_DESCRIPTION,
+            # This defines the different columns of the dataset and their types
+            features=features,  # Here we define them above because they are different between the two configurations
+            # If there's a common (input, target) tuple from the features,
+            # specify them here. They'll be used if as_supervised=True in
+            # builder.as_dataset.
+            supervised_keys=None,
+            # Homepage of the dataset for documentation
+            homepage=_HOMEPAGE,
+            # License for the dataset if available
+            license=_LICENSE,
+            # Citation for the dataset
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        data_dir = dl_manager.download_and_extract(_URL)
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                # These kwargs will be passed to _generate_examples
+                gen_kwargs={
+                    "dirpath": os.path.join(data_dir),
+                },
+            ),
+        ]
+
+    def _generate_examples(
+        self,
+        dirpath,
+    ):
+        """Yields examples as (key, example) tuples."""
+        for id_, xml in enumerate(Path(dirpath).rglob("**/*.xml")):
+            tree = ET.parse(xml)
+            root = tree.getroot()
+            source = tree.findall(".//{http://purl.org/dc/elements/1.1/}source")[0].text
```

Comment:
Also, maybe worth passing a dict to `namespaces=` parameter instead of manually adding the explicit URI to every tag?